Kernel/Devices/VMWareSVGA/VMWareSVGADevice.cpp

```diff
@@ -0,0 +1,286 @@
+            m_framebuffer_max_size = PCI::get_BAR_space_size(address, 1);
+            m_fifo_mapping = MM.allocate_kernel_region(PhysicalAddress(page_base_of(PCI::get_BAR2(address) & 0xfffffff0)), page_round_up(PCI::get_BAR_space_size(address, 2)), "VMWareSVGA Driver FIFO", Region::Access::Read | Region::Access::Write, Region::Cacheable::No);
+            m_fifo_addr = m_fifo_mapping->vaddr();
+            m_interrupt_line = PCI::get_interrupt_line(address);
+            change_irq_number(m_interrupt_line);
+            dbgln("VMWareSVGA: pci address: {}, io_base: {:x}, framebuffer phys: {:x}", address, m_io_base, m_framebuffer_address_physical.as_ptr());
+            // Clear framebuffer memory with arbitrary value
+            OwnPtr<Region> test_map = MM.allocate_kernel_region(m_framebuffer_address_physical, page_round_up(PCI::get_BAR_space_size(address, 1)), "VMWareSVGA FB Test", Region::Access::Write, Region::Cacheable::No);
+            // memset(test_map->vaddr().as_ptr(), 0xCC, PCI::get_BAR_space_size(address, 1));
+            test_map->unmap();
+        }
+    });
+    VERIFY(m_io_base);
+    VERIFY(m_framebuffer_address_physical.as_ptr());
+    VERIFY(m_fifo_addr.as_ptr());
+
+    // Determine SVGA Version
+    // The way this is done ist by writing the highest version we support (2) into a device support register,
+    // and checking if the value has been changed.
+    // If it hasnt, then we found the highest version that both we and the "card" support.
+    // If it has, decrement by one and repeat.
+    m_device_version = make_version_id(2);
+    set_register(IORegister::ID, m_device_version);
+    while (get_register(IORegister::ID) != m_device_version && m_device_version >= make_version_id(0)) {
+        m_device_version--;
+        set_register(IORegister::ID, m_device_version);
+    }
+
+    VERIFY(m_device_version >= make_version_id(0));
+
+    // We want to read the size of the Framebuffer
+    m_fifo_size = get_register(IORegister::MemSize);
+    m_framebuffer_fb_size = get_register(IORegister::MemSize);
+
+    VERIFY(m_fifo_size < 0x20000);
+    VERIFY(m_framebuffer_fb_size < 0x100000);
+
+    // Check if the card is new enough and get the capabilites register
+    if (m_device_version >= make_version_id(1)) {
+        m_device_capabilites = get_register(IORegister::Capabilities);
+    }
+
+    // Check if interrupt masks are supported
+    if (get_capability(Capabilities::IRQMask)) {
+        // Mask all interrupts
+        set_register(IORegister::IRQMask, 0);
+        // Clear all interrupts on the card
+        IO::out32(m_io_base + static_cast<u32>(Ports::IRQStatus), 0xFF);
+        // Enable the interrupt handler
+        enable_irq();
+    }
+
+    dbgln("VMWareSVGA: handshake complete, version {}, max res: {}x{}, irq: {}", m_device_version & 0xFF, get_register(IORegister::MaxWidth), get_register(IORegister::MaxHeight), m_interrupt_line);
+
+    // Set a video mode
+    // set_mode(800, 600, 32);
```

Comment:
is this supposed to still be here?

Reply:
Nope, accidentally left it there. Was trying to mimic the Bochs driver, but this is not necessary.